README.md

```diff
@@ -91,14 +91,14 @@ export const UserContainerModule = CreateModule([
 
 -   **Entity, Provider, Use case, repository, controller examples**: Folders organization for the clean architecture.
 
--   **Error handling**: Using `Report.Error()` we can report errors using predefined error codes. This is useful when we want to report known errors to the client. Inside of `Report.Error()` there is a try catch block encapsulating the error.
+-   **Error handling**: Using `new Report().Error()` we can report errors using predefined error codes. This is useful when we want to report known errors to the client. Inside of `new Report().Error()` there is a try catch block encapsulating the error.
 
 ```typescript
 type ErrorType = GeneralErrorCode | ApplicationErrorCode | HttpStatusErrorCode;
 
 // Reporting errors
 if (userExist) {
-    const error = Report.Error(new ApplicationError(StatusCode.BadRequest, "User already exist!"), "user-create");
+    const error = new Report().Error({ statusCode: StatusCode.BadRequest, message: "User already exist!", service: "user-create"});
```

Comment:
The initial is great, however the user experience using new Report().Error is still a bit odd. The only problem we actually need to solve is that how we can capture the stack trace from where the error happen without constructing an instance of a type?
If we achieve this we can supply to the user a smooth experience like Report.Error(statusCode.BadRequest, "User already created", "user-create-usecase")

Reply:
Done!